### PR TITLE
Add GHRSST SST IODA converter

### LINKED ISSUE #609: GHRSST (AVHRR, VIIRS, does that also include the MW instruments?)

### PR DESCRIPTION
**Pull request by ShastriPaturi**
This PR adds the GHRSST SST IODA converter.
The original data are the L3U from the NCEP data tanks

Fixes #609

### CODE REVIEW

File: utils/obsproc/applications/gdas_obsprovider2ioda.h [hunk 1]
```diff
@@ -26,6 +27,8 @@ namespace gdasapp {
         Rads2Ioda conv2ioda(fullConfig, this->getComm());
         conv2ioda.writeToIoda();
       } else if (provider == "GHRSST") {
+        oops::Log::info() << "GHRSST!" << std::endl;
```
Comment (guillaumevernieres) [inline utils/obsproc/applications/gdas_obsprovider2ioda.h]: look at what's done for the other provider, and remove the log. You clearly have not looked at the output, that code is not writing a ioda file.

File: utils/test/testinput/gdas_ghrsst2ioda.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+provider: GHRSST
+window begin: 2018-04-15T06:00:00Z
+window end: 2018-04-15T12:00:00Z
+variable: seaSurfaceTemperature
+output file: ghrsst_sst_mb_20210701.ioda.nc
+input files:
+- ghrsst_sst_mb_202107010000.nc4
+#- ghrsst_sst_mb_202107010100.nc4
```
Comment (guillaumevernieres) [inline utils/test/testinput/gdas_ghrsst2ioda.yaml]: you need to test the concatenation as well, uncomment line 8.

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
fix the code norm and see comments below.

File: utils/test/testinput/gdas_ghrsst2ioda.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+provider: GHRSST
+window begin: 2018-04-15T06:00:00Z
+window end: 2018-04-15T12:00:00Z
+variable: seaSurfaceTemperature
```
Comment (apchoiCMD) [inline utils/test/testinput/gdas_ghrsst2ioda.yaml:4]: Also remove the variables

File: utils/obsproc/applications/gdas_obsprovider2ioda.h [hunk 4]
```diff
@@ -27,7 +28,8 @@ namespace gdasapp {
         Rads2Ioda conv2ioda(fullConfig, this->getComm());
         conv2ioda.writeToIoda();
       } else if (provider == "GHRSST") {
-        oops::Log::info() << "Comming soon!" << std::endl;
+        Ghrsst2Ioda conv2ioda(fullConfig);
```
Comment (apchoiCMD) [inline utils/obsproc/applications/gdas_obsprovider2ioda.h:31]: missing `this->getComm()` after `fullConfig`

File: utils/test/CMakeLists.txt [hunk 5]
```diff
@@ -2,9 +2,11 @@
 list( APPEND utils_test_input
   testinput/gdas_meanioda.yaml
   testinput/gdas_rads2ioda.yaml
+  testinput/gdas_ghrsst2ioda.yaml
   testinput/gdas_smap2ioda.yaml
   testinput/gdas_smos2ioda.yaml
   testinput/gdas_icecamsr2ioda.yaml
+  testinput/gdas_smos2ioda.yaml
```
Comment (apchoiCMD) [inline utils/test/CMakeLists.txt:9]: conflict `gdas_smos2ioda.yaml`

**Review by apchoiCMD — CHANGES_REQUESTED** [review]

**Comment by guillaumevernieres** [comment]
Closing your PR @ShastriPaturi . I'll submit a PR with your work later today or tomorrow, along with the sub-sampling/superobing tools.

**Pull request closed without merging**